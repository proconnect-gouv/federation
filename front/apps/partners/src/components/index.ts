/* istanbul ignore file */

// declarative file
export * from './items-counter';
export * from './service-provider-header';
export * from './service-provider-sandbox';
export * from './service-providers-list';
export * from './service-providers-list-item';
export * from './service-providers-list-item-detail';
export * from './service-providers-page-title';
