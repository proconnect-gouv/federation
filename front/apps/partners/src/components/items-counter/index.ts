/* istanbul ignore file */

// declarative file
export * from './items-counter.component';
