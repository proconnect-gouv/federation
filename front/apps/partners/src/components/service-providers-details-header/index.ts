/* istanbul ignore file */

// declarative file
export * from './service-providers-details-header.component';
