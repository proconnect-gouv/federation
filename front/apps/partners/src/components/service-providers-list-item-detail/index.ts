/* istanbul ignore file */

// declarative file
export * from './service-providers-list-item-detail.component';
