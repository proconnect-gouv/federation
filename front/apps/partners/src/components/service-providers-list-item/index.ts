/* istanbul ignore file */

// declarative file
export * from './service-providers-list-item.component';
