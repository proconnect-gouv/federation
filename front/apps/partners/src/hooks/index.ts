/* istanbul ignore file */

// declarative file
export * from './add-service-provider-config';
export * from './get-service-provider-configs';
export * from './service-provider-details';
export * from './service-provider-sandbox';
