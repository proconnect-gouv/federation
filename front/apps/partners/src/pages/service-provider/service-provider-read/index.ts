/* istanbul ignore file */

// declarative file
export * from './service-provider-read.page';
