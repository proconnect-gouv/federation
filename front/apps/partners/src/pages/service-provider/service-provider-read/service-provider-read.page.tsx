import React from 'react';
import { useParams } from 'react-router-dom';

import { ServiceProviderHeaderComponent } from '../../../components';
import { ServiceProvidersActionTypes } from '../../../enums';
import { useServiceProviderDetails } from '../../../hooks';

export const ServiceProviderReadPage = React.memo(() => {
  const { id } = useParams() as { id: string };
  const { item } = useServiceProviderDetails({
    id,
    type: ServiceProvidersActionTypes.SERVICE_PROVIDER_READ_REQUESTED,
  });

  return (
    <div>
      {item && (
        <div className="fr-container fr-mt-3w">
          <ServiceProviderHeaderComponent item={item} />
        </div>
      )}
    </div>
  );
});

ServiceProviderReadPage.displayName = 'ServiceProviderReadPage';
