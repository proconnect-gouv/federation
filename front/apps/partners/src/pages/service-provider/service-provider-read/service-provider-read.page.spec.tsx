import { render } from '@testing-library/react';
import { useParams } from 'react-router-dom';

import { ServiceProviderHeaderComponent } from '../../../components';
import { ServiceProviderStatusColors } from '../../../enums';
import { useServiceProviderDetails } from '../../../hooks';
import { ServiceProviderReadPage } from './service-provider-read.page';

jest.mock('react-router-dom');
jest.mock('../../../hooks/service-provider-details/service-provider-details.hook');
jest.mock('../../../components/service-provider-header/service-provider-header.component');

const itemMock = {
  color: ServiceProviderStatusColors.SANDBOX,
  platformName: 'chaine traduite',
  spName: 'name',
  status: 'chaine traduite',
};

describe('ServiceProviderReadPage', () => {
  beforeEach(() => {
    // given
    jest.mocked(useParams).mockReturnValueOnce({ id: '42' });
  });

  it('should match the snapshot', () => {
    // given
    jest.mocked(useServiceProviderDetails).mockReturnValueOnce({
      item: itemMock,
    });

    // when
    const { container } = render(<ServiceProviderReadPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should call hook function useServiceProviderDetails', () => {
    // given
    jest.mocked(useServiceProviderDetails).mockReturnValueOnce({
      item: undefined,
    });

    // when
    render(<ServiceProviderReadPage />);

    // then
    expect(useServiceProviderDetails).toHaveBeenCalledTimes(1);
  });

  it('should call ServiceProviderHeaderComponent when item is defined', () => {
    // given
    jest.mocked(useServiceProviderDetails).mockReturnValueOnce({
      item: itemMock,
    });

    // when
    render(<ServiceProviderReadPage />);

    // then
    expect(ServiceProviderHeaderComponent).toHaveBeenCalledTimes(1);
  });

  it('should not call ServiceProviderHeaderComponent when item is undefined', () => {
    // given
    jest.mocked(useServiceProviderDetails).mockReturnValueOnce({
      item: undefined,
    });

    // when
    render(<ServiceProviderReadPage />);

    // then
    expect(ServiceProviderHeaderComponent).not.toHaveBeenCalled();
  });
});
