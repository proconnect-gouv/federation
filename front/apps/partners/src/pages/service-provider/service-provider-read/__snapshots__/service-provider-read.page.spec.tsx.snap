// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceProviderReadPage should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="fr-container fr-mt-3w"
    >
      <div>
        ServiceProviderHeaderComponent
      </div>
    </div>
  </div>
</div>
`;
